android/CouchbaseLite/src/androidTest/java/com/couchbase/lite/ArrayTest.java

```diff
@@ -156,16 +156,16 @@ public void validate(Array a) {
                 assertEquals(true, a.getObject(0));
                 assertEquals(false, a.getObject(1));
                 assertEquals("string", a.getObject(2));
-                assertEquals(0, a.getObject(3));
-                assertEquals(1, a.getObject(4));
-                assertEquals(-1, a.getObject(5));
+                assertEquals(0, ((Number)a.getObject(3)).intValue());
+                assertEquals(1, ((Number)a.getObject(4)).intValue());
+                assertEquals(-1, ((Number)a.getObject(5)).intValue());
                 assertEquals(1.1, a.getObject(6));
                 assertEquals(kArrayTestDate, a.getObject(7));
                 assertEquals(null, a.getObject(8));
 
                 // dictionary
                 Dictionary subdict = (Dictionary) a.getObject(9);
-                Map<String, Object> expectedMap = new HashMap<>();
+                Map<String, Object> expectedMap = new TreeMap<>();
```

Comment:
TreeMap is easier for testing as the order of keys is guaranteed.
I can switch it back.